broker/client.go

```diff
@@ -94,7 +94,12 @@ var (
 func (c *client) init() {
 	c.status = Connected
 	c.info.localIP, _, _ = net.SplitHostPort(c.conn.LocalAddr().String())
-	c.info.remoteIP, _, _ = net.SplitHostPort(c.conn.RemoteAddr().String())
+	remoteAddr := c.conn.RemoteAddr()
+	remoteNetwork := remoteAddr.Network()
+	c.info.remoteIP = ""
+	if remoteNetwork != "websocket" {
+		c.info.remoteIP, _, _ = net.SplitHostPort(remoteAddr.String())
```

Comment:
`remoteAddr.String()` runs into a nil pointer exception for websocket connections (the URL seems not to be set). Working around by not setting `remoteIP` in this case.